src/geojson-layer.ts

```diff
@@ -169,11 +214,31 @@ export default class GeoJSONLayer extends React.Component<Props, {}> {
           const layoutDiff = diff(this.props[prop], props[prop]);
 
           Object.keys(layoutDiff).forEach(key => {
-            map.setLayoutProperty(`${this.id}-${type}`, key, layoutDiff[key]);
+            map.setLayoutProperty(this.buildLayerId(type), key, layoutDiff[key]);
           });
         }
       }
     }
+
+    for (const type in typeToLayerLUT) {
+      if (typeToLayerLUT.hasOwnProperty(type)) {
+        for (const event in eventToHandler) {
+          if (eventToHandler.hasOwnProperty(event)) {
+            const prop = typeToLayerLUT[type] + eventToHandler[event];
+
+            if (!isEqual(props[prop], this.props[prop])) {
```

Comment:
To test equality on a function I think you can go for a reference check straight away so `props[prop] === this.props[prop]` no need for value equality and `===` is much faster